onnxruntime/core/providers/nuphar/scripts/symbolic_shape_infer.py

```diff
@@ -850,15 +852,30 @@ def _infer_Reshape(self, node):
 
             assert new_sympy_shape.count(-1) < 2
             if -1 in new_sympy_shape:
-                new_dim = total // non_deferred_size
+                # handle a special case where input_shape is [] and shape_value is [-1]
+                if len(input_shape) == 0 and len(shape_value) == 1:
+                    if node.output[0] in self.sympy_data_:
+                        sympy_shape = self.sympy_data_[node.output[0]][0]
+                    else:
+                        # create the sympy data if not exist
+                        self._pass_on_sympy_data(node)
+                        skip_pass_on_sympy_data = True
+                        if node.output[0] in self.sympy_data_:
+                            sympy_shape = self.sympy_data_[node.output[0]][0]
+                        else:
+                            sympy_shape = self._new_symbolic_dim_from_output(node)
```

Comment:
>sympy_shape = self._new_symbolic_dim_from_output(node [](start = 28, length = 53)

Isn't the shape empty?

Reply:
Did you mean that we don't need to create a new symbolic dim in the ```else``` branch? Or, you meant that we don't need a symbolic dim inside the ```True``` branch of ```if len(input_shape) == 0 and len(shape_value) == 1```?